### ISSUE TITLE
Allow local config / download + read JDK path from the environment

### ISSUE THREAD

**Issue by toastal** [GoogleChromeLabs/bubblewrap#457]
**Is your feature request related to a problem? Please describe.**
All of the docs suggest `npm install --global ...` and the config is read from user home. If you wanted to install this tool locally to keep versions pinned (both for the CLI tool as well as the optional automatic JDK) for a specific project, you can't really. Being able to access the local config would also make this more accessible towards using tools like `nix shell` to distribute and source the JDK version. Further useful would be the ability to use the env vars to point to the JDK version so something like `direnv` could run a command `which` or `nix-build --no-out-link '<nixpkgs>' -A jdk8_headless` to find the JDK version.

**Describe the solution you'd like**
I would like to see a first-local fallback for `.bubblewrap/config` file + something like `$JAVA_HOME` or `$JDK_HOME` that could be read could override the current tooling. Alternatively, allowing the ability to pass in the config file via command line arguments could be an easy solution. I think there's now too much magic in downloading Java (like assuming all Linuxes are x86_64, etc.) that you made it too difficult to override if using a different setup.

**Describe alternatives you've considered**
I'm currently trying to pull apart `@bubblewrap/core` and override the bits I need in a custom command. This is far from ideal as I'm already starting at a bad place with the node v14.15.4 REPL 

```js
var core = require("@bubblewrap/core")
```

outputting 

```sh
Uncaught Error: libuuid.so.1: cannot open shared object file: No such file or directory
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1122:18)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
```

_EDIT:_ I actually get the same error output as above just by running the `node_modules/.bin/bubblewrap`. It would seem there's a mess of missing dependencies for Node canvas ... which we need why?

**Additional context**
Thanks for putting the tool together. I think it's nice for a lot of people that ya'll are handling the JDK version management on generic Ubuntu user's laptops. There's instructions for how to override and manually install the dependencies, but a) they're set to global and b) they could/should be easier to override.

**Comment by andreban** [comment]
Automatically downloading the JDK / Android SDK was added because it was, by far, the biggest source of issues from users.  One of the reasons is that the Android tools require JDK 8 and anything different will throw errors, which becomes error prone. We also ran into new releases of the Android SDK being broken, so downloading a version that is known to work well helps.

Regarding 32 bit support, the Android SDK tools [requires 64 bit Linux distribution](https://developer.android.com/studio#system-requirements-a-namerequirementsa), so we have to assume the machine is x86_64. It sounds to me the best solution is to throw an error message here.

I'm supportive of enabling users to set a custom configuration, either with a command-line parameter or environment variables. I'm not a big fan of `JAVA_HOME` or similar, as it's likely the user will have a JDK that's different from version 8, which is required by the Android SDK and will cause troubles, but could be convinced differently.

We have an [issue open](https://github.com/GoogleChromeLabs/bubblewrap/issues/362) around adhering to the [XDG specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), which would help with where to look for the config files, by setting the `XDG_CONFIG_HOME` environment variable. 

This [other issue](https://github.com/GoogleChromeLabs/bubblewrap/issues/243) is also about looking for existing installations of the JDK/Android SDK.

> I'm currently trying to pull apart @bubblewrap/core and override the bits I need in a custom command. This is far from ideal as I'm already starting at a bad place with the node v14.15.4 REPL

Alternatively, we'd be happy to accept pull-requests that would make configurability better. :wink:. [`loadOrCreateConfig()`](https://github.com/GoogleChromeLabs/bubblewrap/blob/master/packages/cli/src/lib/config.ts#L93-L94) can take a custom path when constructed, so [`Cli.ts`](https://github.com/GoogleChromeLabs/bubblewrap/blob/master/packages/cli/src/lib/Cli.ts#L39) could be modified to read an environment variable that uses that.

>EDIT: I actually get the same error output as above just by running the node_modules/.bin/bubblewrap. It would seem there's a mess of missing dependencies for Node canvas ... which we need why?

This is pulled in by [svg2img](https://www.npmjs.com/package/svg2img) so we can handle SVGs, which are frequently used in Web Manifests. This is the only native dependency in Bubblewrap, but we couldn't find a pure JS SVG library. They are all browser libraries, which use the browser SVG support or Node.js libraries that use `canvas`.  I had a [draft PR around optionally supporting svgs](https://github.com/GoogleChromeLabs/bubblewrap/pull/378), but we didn't merge it (not sure why right now). Again, maybe worth taking a fresh look into optionally loading the svg library. I recently created [an issue](https://github.com/GoogleChromeLabs/bubblewrap/issues/456) for failing builds on Chrome OS / Crostini, with a workaround. My guess is you'll need to install missing libraries.

**Comment by toastal** [comment]
Right that's what I was seeing. I think being on full NixOS brings a lot of unforeseen pain because a lot of tools expect a different behavior -- like how most people's XDG points at their home dir so potato pot_a_to -- but there are many differences tooling author's don't have the time/resources for _not_ assuming Linux is glibc on x86_64. Regardless, I'll see what I can do. It's a shame I can't get the node canvas running because it's _not_ an optional dependency so I could try it out. I think with that being optional, I could use the parts of core I want without duplicating too much effort.

**Comment by andreban** [comment]
#461 has published with [v1.10.0](https://github.com/GoogleChromeLabs/bubblewrap/releases/tag/v1.10.0). @toastal does this solve this issue?

**Comment by toastal** [comment]
Yeah, I think this can make it workable for a lot of situations.
